controllers/commentController.js

```diff
@@ -19,3 +24,28 @@ exports.getArtCommentsById = (request, response, next) => {
       next(error);
     });
 };
+
+
+exports.postArtCommentById = (request, response, next) => {
+  const { article_id } = request.params;
+  const { username, body } = request.body;
+
+  selectArticleById(article_id)
+    .then((result) => {
+      // if we have a result here, don't care, as it hasn't already been rejected
+      // article must exist, so:
+      // check for author next
+      return selectUsernameByName(username); // return or reject in model
+    }) 
+    .then((result) => { 
+
+      return insertCommentByArtId(article_id, username, body); // return or reject in model
+    })
+    .then((result) => {
+      
+      response.status(201).send({ comment: result[0] }); // respond with the newly inserted row.
```

Comment:
For better code, result[0] could just be result and you would make sure to send the correct data back from your model